app/services/hyku_addons/collection_branding_fetcher.rb

```diff
@@ -2,13 +2,14 @@
 
 module HykuAddons
   class CollectionBrandingFetcher
-    def self.collection_banner(id)
+    def collection_banner(id, cname)
+      AccountElevator.switch!(cname)
       @_collection_banner ||= collection_branding_info(id).where(role: 'banner')
     end
 
     private_class_method
 
-    def self.collection_branding_info(id)
+    def collection_branding_info(id)
       @_collection_branding_info ||= CollectionBrandingInfo.where(collection_id: id.to_s)
     end
```

Comment:
So, i love memoizing complex functions, but is it worth it in this case? 

`CollectionBrandingFetcher` is only used in one json file and its assigned to a variable after which, the variable is used for the rest of the request. This just means that you will be storing the item in memory on the instance, then assigning it to another variable, so using double the memory that you might otherwise use if you weren't memoizing it. 

I think memoizing variables is great if you were calling the method multiple times, but as its only called once, its probably best to remove the extra compexity.

Reply:
I think this is memoized because the query is complex and what is memoized in each is a different query response, so it is not thesame  data that is in memory

Reply:
Sure, memoizing queries is a great idea, but this one only ever gets run one and then it assigned to a variable in the view, so the query is run, its stored on the instance, assigned to the view variable and that is accessed from then on. So there is never a secondary call of the method to make the memoization worthwhile.

Reply:
I will remove one them.